lib/src/algorithms.dart

```diff
@@ -32,10 +32,10 @@ int _comparableBinarySearch/*<T extends Comparable<T>>*/(
 /// the objects.
 ///
 /// Returns -1 if [value] is not in the list by default.
-int binarySearch/*<T extends Comparable<T>>*/(
+int binarySearch/*<T>*/(
     List/*<T>*/ sortedList, /*=T*/ value, { int compare(/*=T*/ a, /*=T*/ b) }) {
   if (compare == null) {
-    return _comparableBinarySearch(sortedList, value);
+    return _comparableBinarySearch(sortedList as List<Comparable>, value as Comparable);
```

Comment:
Yes, I understand.  I just ran into this issue when we updated package:collection and the analyzer turned red because we were calling this function with a List<Foo> where Foo didn't implement Comparable.  Previously that was permitted because this function takes a `compare` argument to compare object that don't implement `Comparable`.  Currently that's disallowed by the analyzer.  I'm not sure whether changes that break only the analyzer count as breaking changes for semantic versioning, but they turn our buildbot red.

I'm not sure what the intent of the package:collection authors is here.  I suspect it might be time for me to step back and let @nex3 or whoever decide what they think is best.

I guess I could file an issue about how package:collection version 1.4.1 breaks analyzer compatbility with version 1.4.0, but there isn't really anything that can be done about that issue anymore because version 1.4.1 has been published to pub and cannot be rolled back.